pkg/snapstore/abs_snapstore_test.go

```diff
@@ -7,286 +7,163 @@ package snapstore_test
 import (
 	"bytes"
 	"context"
-	"encoding/xml"
+	"encoding/base64"
 	"fmt"
 	"io"
-	"net/http"
-	"net/url"
 	"sort"
-	"strconv"
 	"strings"
 	"sync"
 
-	"github.com/Azure/azure-pipeline-go/pipeline"
-	"github.com/Azure/azure-storage-blob-go/azblob"
-	. "github.com/onsi/gomega"
-
-	. "github.com/gardener/etcd-backup-restore/pkg/snapstore"
-	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blockblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
 )
 
-func newFakeABSSnapstore() brtypes.SnapStore {
-	f := []pipeline.Factory{
-		pipeline.MethodFactoryMarker(),
-		newFakePolicyFactory(bucket, prefixV2, objectMap),
-	}
-	p := pipeline.NewPipeline(f, pipeline.Options{HTTPSender: newFakePolicyFactory(bucket, prefixV2, objectMap)})
-	u, err := url.Parse(fmt.Sprintf("https://%s.%s", "dummyaccount", brtypes.AzureBlobStorageGlobalDomain))
-	Expect(err).ShouldNot(HaveOccurred())
-	serviceURL := azblob.NewServiceURL(*u, p)
-	containerURL := serviceURL.NewContainerURL(bucket)
-	a, err := GetABSSnapstoreFromClient(bucket, prefixV2, "/tmp", 5, brtypes.MinChunkSize, &containerURL)
-	Expect(err).ShouldNot(HaveOccurred())
-	return a
+type fakeABSContainerClient struct {
+	objects map[string]*[]byte
+	prefix  string
+	mutex   sync.Mutex
+	// a map of blobClients so new clients created to a particular blob refer to the same blob
+	blobClients map[string]*fakeBlockBlobClient
```

Comment:
do you want to mock the AzureBlockBlob Client ?
```
//go:generate mockgen -package <name> -destination=mocks.go github.com/gardener/etcd-backup-restore/pkg/snapstore/ AzureBlockBlobClientI
```

Reply:
I will leave this to another PR. Will create an issue to track this later.

Reply:
can you create an issue for it and mentioned it here.